extra_data/tests/test_reader_mockdata.py

```diff
@@ -168,6 +188,17 @@ def test_iterate_spb_raw_run(mock_spb_raw_run):
     assert device in data
     assert data[device]['data.image.pixels'].shape == (1024, 768)
 
+
+def test_iterate_spb_raw_run_keep_dims(mock_spb_raw_run):
+    run = RunDirectory(mock_spb_raw_run)
+    trains_iter = run.trains(keep_dims=True)
```

Comment:
Is it worth doing a `.select()` in here? Even with no real data, creating all the arrays and making all the calls into HDF5 might be a bit slow.